pages/packingDetailed.js

```diff
@@ -0,0 +1,333 @@
+import Layout from "../components/Layout";
+import Head from "next/head";
+import useSWR from "swr";
+import cookie from "js-cookie";
+import { useRouter } from 'next/router'
+import {
+  ORDER_STATUS_COMPLETE,
+  ORDER_STATUS_OPEN,
+  ORDER_STATUS_PROCESSING,
+} from "../utils/orderStatuses";
+
+const token = cookie.get("firebaseToken");
+
+function fetcher(...urls) {
+  const f = (u) =>
+    fetch(u, {
+      headers: { "Content-Type": "application/json", Authorization: token },
+    }).then((r) => r.json());
+
+  if (urls.length > 1) {
+    return Promise.all(urls.map(f));
+  }
+  return f(urls);
+}
+
+class PackingOrder extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      orderId: props.data[0].orderId,
+      delivery_date: props.data[0].delivery_date,
+      pantryNote: props.data[0].pantry_note,
+      guestNote: props.data[0].guestNote,
+      firstName: props.data[0].firstName,
+      lastInitial: props.data[0].lastInitial,
+      status: props.data[0].status,
+      order_timestamp: props.data[0].order_timestamp,
+      items: props.data[0].items,
+      itemMap: props.data[1],
+      error: null,
+      success: null,
+    };
+  }
+
+  savePantryNote = (newPantryNote = this.state.pantryNote) => {
+    this.state.pantryNote = newPantryNote;
+    this.setState({ pantryNote: this.state.pantryNote });
+    fetch("/api/orders/SetPantryNote", {
+      method: "POST",
+      body: JSON.stringify({
+        orderId: this.state.orderId,
+        message: this.state.pantryNote,
+      }),
+      headers: { "Content-Type": "application/json", Authorization: token },
+    }).then(() => {
+      this.setState({ success: "Saved pantry note successfully!" });
+      setTimeout(() => this.setState({ success: null }), 1000);
+    });
+  };
+
+  cancelPantryNote = () => {
+    document.getElementById("pantry_note").value = this.state.pantryNote;
+  };
+
+  changeOrderStatus = () => {
+    var newStatus = "";
+    if (this.state.status == ORDER_STATUS_OPEN) {
+      newStatus = ORDER_STATUS_PROCESSING;
+    } else if (this.state.status == ORDER_STATUS_PROCESSING) {
+      newStatus = ORDER_STATUS_COMPLETE;
+    } else {
+      newStatus = ORDER_STATUS_PROCESSING;
+    }
+    this.state.status = newStatus;
```

Comment:
I think you should use `this.setState` to modify local state, rather than `this.state.status = ...`

Reply:
ya u should be able to just put newStatus directly into `this.setState({ status: newStatus })`